#! /bin/sh /usr/share/dpatch/dpatch-run
## 95_charset-vdr-1.5.3.dpatch by Matthias Schwarzott <zzam@gentoo.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes utf8 issues with vdr >= 1.5.3

@DPATCH@
diff -urNad vdr-plugin-burn-0.1.0~pre21~/burn.c vdr-plugin-burn-0.1.0~pre21/burn.c
--- vdr-plugin-burn-0.1.0~pre21~/burn.c	2006-09-16 20:33:36.000000000 +0200
+++ vdr-plugin-burn-0.1.0~pre21/burn.c	2008-05-11 14:51:07.000000000 +0200
@@ -45,8 +45,13 @@
 
 	string plugin::get_character_encoding()
 	{
-#ifdef UTF8PATCH
-		return "utf8";
+#if VDRVERSNUM >= 10503
+		if (cCharSetConv::SystemCharacterTable())
+			return cCharSetConv::SystemCharacterTable();
+		else
+			return "utf-8";
+#elif defined(UTF8PATCH)
+		return "utf-8";
 #else
 		return tr("iso8859-15");
 #endif
